src/components/ebay-combobox/index.js

```diff
@@ -0,0 +1,179 @@
+            options: []
+        }, input);
+    },
+    getInitialState(input) {
+        const index = findIndex(input.options, option => option.selected);
+        const currentValue = input.options[index] && input.options[index].text;
+
+        return Object.assign({}, input, {
+            selectedIndex: index === -1 ? null : index,
+            currentValue: currentValue
+        });
+    },
+    init() {
+        observer.observeRoot(this, ['disabled'], () => {
+            this.expander.expandOnClick = !this.state.disabled;
+        });
+
+        this.getEls('option').forEach((optionEl, i) => {
+            Object.defineProperty(optionEl, 'selected', {
+                get: () => this.state.selectedIndex === i,
+                set: (value) => this.setSelectedIndex(value ? i : null)
+            });
+        });
+    },
+    onRender() {
+        const selectedIndex = findIndex(this.state.options, option => option.selected);
+
+        if (!this.state.disabled && this.state.options.length) {
+            this.activeDescendant = ActiveDescendant.createLinear(
+                this.el,
+                this.getEl('input'),
+                this.getEl('options'),
+                '.combobox__option[role=option]', {
+                    activeDescendantClassName: 'combobox__option--active',
+                    autoInit: selectedIndex === -1 ? -1 : 0,
+                    autoReset: -1
+                }
+            );
+
+            this.expander = new Expander(this.el, {
+                autoCollapse: true,
+                expandOnFocus: true,
+                expandOnClick: this.state.readonly && !this.state.disabled,
+                collapseOnFocusOut: !this.state.readonly,
+                contentSelector: '.combobox__options',
+                hostSelector: '.combobox__control > input',
+                expandedClass: 'combobox--expanded',
+                simulateSpacebarClick: true
+            });
+        }
+
+        this.moveCursorToEnd();
+    },
+    onBeforeUpdate() {
+        this._handleDestroy();
+    },
+    onDestroy() {
+        this._handleDestroy();
+    },
+    _handleDestroy() {
+        if (this.activeDescendant) {
+            this.activeDescendant.destroy();
+        }
+
+        if (this.expander) {
+            this.expander.cancelAsync();
+        }
+    },
+    handleExpand() {
+        elementScroll.scroll(this.getEls('option')[this.state.selectedIndex]);
+        this.moveCursorToEnd();
+        emitAndFire(this, 'combobox-expand');
+    },
+    handleCollapse() {
+        emitAndFire(this, 'combobox-collapse');
+    },
+    moveCursorToEnd() {
+        const currentInput = this.getEl('input');
+
+        if (currentInput) {
+            const len = currentInput.value.length;
+            currentInput.setSelectionRange(len, len);
+        }
+    },
+    handleComboboxKeyUp(originalEvent) {
+        const selectedEl = this.getEl('options').querySelector('.combobox__option--active');
```

Comment:
To avoid using a querySelector, I'd prefer to see:
```
const selectedEl = this.getEls('option')[this.state.selectedIndex];
```

Reply:
But, this doesn't change util the the option is 'clicked'. E.g. makeup-activedescendant doesn't change the `selectedIndex`.